plugins/cost-insights/src/components/CostInsightsPage/CostInsightsPage.tsx

```diff
@@ -48,15 +55,17 @@ const CostInsightsPage = () => {
   // There is not currently a UI to set feature flags
   // flags.set('cost-insights-currencies', FeatureFlagState.On);
   const client = useApi(costInsightsApiRef);
-  const { currencies } = useConfig();
+  const config = useConfig();
```

Comment:
Do we need this change?

Reply:
Artifact from a previous approach :/ No. 

The drawback of destructuring is that it introduces more variables into the local namespace, which can produce collisions and require weird workarounds. Plus, it looks a little cleaner since the neighboring hooks aren't destructured.

One one hand, it doesn't get any cleaner than writing `currencies={currencies}` but it can be more descriptive and helpful to see `currencies={config.currencies}` - so we know exactly where that value is coming from. (This is particular true when there is overlap in property names. (see: `filter.metric` and `config.metrics`, etc.)

Reply:
Yeah I agree w/ the approach, I think we'll just have to try to keep it consistent. I think there's a couple other places where we destructure.

Reply:
Yeah, doesn't have to be hard rule though. Maybe only in cases where we destructure for a single prop or there's potential for overlap? 

In my mind I follow a rough set of rules:

Single get/set: use array destructuring ala React. Ex:

```
const [foo, setFoo] = useFoo(null);
```

Multiple get/setters: use obj destructuring:

```
const { foo, bar, setFoo, setBar } = useFooBar(null); 
```

No destructuring where there is potential for naming conflicts/overlap, or the props are fairly generic:

```
const foo = useFoo();
const bar = useBar();

foo.metric
bar.metric
```

Reply:
Got it, yeah that makes sense 👍🏼